src/client.rs

```diff
@@ -60,41 +63,22 @@ impl<T, B> Client<T, B>
     ///
     /// It's important to note that this does not **flush** the outbound
     /// settings to the wire.
-    pub fn handshake2(io: T) -> Handshake<T, B> {
+    pub fn handshake2(io: T, config: Config) -> Handshake<T, B> {
```

Comment:
I think I would either call it `Config` and pass it as an arg here (and remove `handshake2` on `Config`) **or** call it `Builder` and remove `pub` from this fn.

I think that I prefer the builder approach personally.